### PR TITLE
Add more logging message to monitor branch token update

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Add more logging message to monitor branch token update

**Why?**
better debugging.

**How did you test it?**
only logging.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/api/get_workflow_util.go [hunk 1]
```diff
@@ -94,6 +96,10 @@ func GetOrPollMutableState(
 	// We return the full version histories. Callers need to fetch the last version history item from current branch
 	// and use the last version history item in following calls.
 	if !versionhistory.ContainsVersionHistoryItem(currentVersionHistory, request.VersionHistoryItem) {
+		logger.Warn("Request history branch and current history branch are mismatched.",
```
Comment (stephanos) [inline service/history/api/get_workflow_util.go:99]:
```suggestion
		logger.Warn("Request history branch and current history branch don't match",
```

File: service/history/api/get_workflow_util.go [hunk 2]
```diff
@@ -121,6 +127,10 @@ func GetOrPollMutableState(
 			return nil, err
 		}
 		if !versionhistory.ContainsVersionHistoryItem(currentVersionHistory, request.VersionHistoryItem) {
+			logger.Warn("Request history branch and current history branch are mismatched prior to poll the mutable state.",
```
Comment (stephanos) [inline service/history/api/get_workflow_util.go:130]:
```suggestion
			logger.Warn("Request history branch and current history branch don't match prior to polling the mutable state",
```

File: service/history/api/get_workflow_util.go [hunk 3]
```diff
@@ -152,6 +162,10 @@ func GetOrPollMutableState(
 				response.CurrentBranchToken = latestVersionHistory.GetBranchToken()
 				response.VersionHistories = event.VersionHistories
 				if !versionhistory.ContainsVersionHistoryItem(latestVersionHistory, request.VersionHistoryItem) {
+					logger.Warn("Request history branch and current history branch are mismatched after polled the mutable state.",
```
Comment (stephanos) [inline service/history/api/get_workflow_util.go:165]:
```suggestion
					logger.Warn("Request history branch and current history branch don't match after polling the mutable state",
```

File: service/history/api/get_workflow_util.go [hunk 4]
```diff
@@ -94,6 +96,10 @@ func GetOrPollMutableState(
 	// We return the full version histories. Callers need to fetch the last version history item from current branch
 	// and use the last version history item in following calls.
 	if !versionhistory.ContainsVersionHistoryItem(currentVersionHistory, request.VersionHistoryItem) {
+		logger.Warn("Request history branch and current history branch are mismatched.",
+			tag.Value(currentVersionHistory),
```
Comment (stephanos) [inline service/history/api/get_workflow_util.go:100]: I don't know much about this yet; is `currentVersionHistory` safe to log? Safe in terms of (a) security, (b) size and (c) performance?

**Review by stephanos — APPROVED** [review]

File: service/history/api/get_workflow_util.go (hunk 4 above)
Comment (yux0) [reply service/history/api/get_workflow_util.go]: change to log the last item. The length is fixed.

File: service/history/api/get_workflow_util.go [hunk 5]
```diff
@@ -121,6 +131,14 @@ func GetOrPollMutableState(
 			return nil, err
 		}
 		if !versionhistory.ContainsVersionHistoryItem(currentVersionHistory, request.VersionHistoryItem) {
+			logItem, err := versionhistory.GetLastVersionHistoryItem(currentVersionHistory)
+			if err != nil {
+				return nil, err
+			}
+			logger.Warn("Request history branch and current history branch don't mismatch prior to poll the mutable state.",
```
Comment (stephanos) [inline service/history/api/get_workflow_util.go:138]:
```suggestion
			logger.Warn("Request history branch and current history branch don't match prior to polling the mutable state",
```

**Review by stephanos — APPROVED** [review]
1 of the 3 messages hasn't been updated yet

**Pull request merged**